src/Build/BackEnd/BuildManager/CacheSerialization.cs

```diff
@@ -61,6 +62,27 @@ public static string SerializeCaches(IConfigCache configCache, IResultsCache res
                             break;
                     }
 
+                    // Avoid creating new config and results caches if no projects were built in violation
+                    // of isolation mode.
+                    if (configCacheToSerialize.Count() > 1)
```

Comment:
```suggestion
                    if (configCacheToSerialize.Count > 0)
```

? "no projects were built" sounds like 0

Reply:
Projects built in violation of isolation mode are those whose results were not entirely provided via the input caches. For example, if we're building in isolation a project A which depends on project B but we only have a strict subset S' of the required target results S from project B provided via input caches, then we build the complement S'' of S'  _in violation of isolation mode_, which means `configCacheToSerialize.Count` would return `2`. (Our config cache would contain entries for A and B.) If S was provided via the input caches, then `configCacheToSerialize.Count` would return `1`, meaning no isolation violation occurred, there are no cache results to exclude, and there is no need to create new config and results caches (i.e., we can just defer to the old serialization logic).

Reply:
Ok, so we _always_ have the project being built in the configCacheToSerialize, but we only _add_ another project if we tried to find it in the cache, failed, and ended up building it separately. I think that makes sense. Thanks!